lake-primitives/src/types/block.rs

```diff
@@ -75,40 +74,15 @@ impl Block {
     /// A reminder that `receipt_execution_outcomes` has a type [near_indexer_primitives::IndexerExecutionOutcomeWithReceipt] which is an
     /// ephemeral structure from `near-indexer-primitives` that hold a [near_primitives::views::ExecutionOutcomeView]
     /// along with the corresponding [near_primitives::views::ReceiptView].
-    pub fn receipts(&mut self) -> impl Iterator<Item = &receipts::Receipt> {
-        if self.executed_receipts.is_empty() {
-            self.executed_receipts = self
-                .streamer_message
-                .shards
-                .iter()
-                .flat_map(|shard| shard.receipt_execution_outcomes.iter())
-                .map(Into::into)
-                .collect();
-        }
+    pub fn receipts(&self) -> impl Iterator<Item = &receipts::Receipt> {
```

Comment:
Is there a strong need to return an iterator? I would just return a slice whenever possible:

```suggestion
    pub fn receipts(&self) -> &[receipts::Receipt] {
        &self.executed_receipts
```